banzai_nres/qc/qc_wavelength.py

```diff
@@ -2,48 +2,109 @@
     @author:mjohnson
     """
 
     def __init__(self, runtime_context):
         super(AssessWavelengthSolution, self).__init__(runtime_context)
 
     def do_stage(self, image):
+        Dlambda_match_threshold = 0.1  # Angstroms
         lab_lines = find_nearest(image.features['wavelength'], np.sort(image.line_list))
-        Delta_lambda = self.calculate_delta_lambda(image,lab_lines)
-        sigma_Dlambda, good_sigma_Dlambda, raw_chi_squared, good_chi_squared = self.calculate_1d_metrics(image, Delta_lambda)
-        x_diff_Dlambda, order_diff_Dlambda = self.calculate_2d_metrics(image,Delta_lambda)
-        qc_results = {'sigma_Dlambda': sigma_Dlambda, 'good_sigma_Dlambda': good_sigma_Dlambda,
-                        'raw_chi_squared': raw_chi_squared, 'good_chi_squared': good_chi_squared,
-                        'Delta_lambda': Delta_lambda, 'x_diff_Dlambda': x_diff_Dlambda,
-                        'order_diff_Dlambda': order_diff_Dlambda}
+        Dlambda = self.calculate_delta_lambda(image, lab_lines)
+        result = self.calculate_1d_metrics(image, Dlambda, lab_lines, Dlambda_match_threshold)
+        sigma_Dlambda, matched_sigma_Dlambda, chi2, matched_chi2, num_matched_lines, velocity_precision = result
+        #x_diff_Dlambda, order_diff_Dlambda = self.calculate_2d_metrics(image, Dlambda)
+        # TODO add the number of overlaps matched during wavelength calibration to this metric.
+        #  the number of fit overlaps may be saved in the features header, check this.
+        qc_results = {'sigma_Dlambda': sigma_Dlambda,
+                      'matched_sigma_Dlambda': matched_sigma_Dlambda,
+                      'wavecal_precision(m/s)': velocity_precision.to(units.meter/units.second).value,
+                      'chi_squared': chi2,
+                      'matched_chi_squared': matched_chi2,
+                      'num_matched_lines': num_matched_lines}
+        qc_description = {'sigma_Dlambda': 'Wavecal statistic. Standard deviation of the distribution of '
+                                           'lab minus measured wavelength residuals',
+                          'matched_sigma_Dlambda': 'Wavecal statistic. sigma_Dlambda but only using lines that agree with a '
+                                                   f'lab line to within {Dlambda_match_threshold} Angstroms.',
+                          'wavecal_precision(m/s)': 'matched_sigma_Dlambda/sqrt(N) in velocity space '
+                                                    '(i.e. standard deviation of delta lambda/lambda * c divided by '
+                                                    'the square root of the number of matched lines) '
+                                                    'for matched lines only. I.e. this is the formal error on the mean '
+                                                    'of the residuals distribution (in velocity units). '
+                                                    'Units of meter per second.',
```

Comment:
change description to something simple. "Measure of dispersion of the wavelength solution in m/s"